pkg/retriever/api.go

```diff
@@ -80,3 +83,31 @@ func (c *CoreRetriever) getCores() {
 		prom.C.SetTotalCoreCount(service.Account, service.Region, float64(totalCoreCount))
 	}
 }
+
+func (c *ApiRetriever) getSpotInstances() {
+	spotRequestItems := []prom.SpotRequestItem{}
+	for _, service := range c.Services {
+		logrus.Infof("Getting SpotInstances for account '%s' and region '%s'", service.Account, service.Region)
+		resp, err := service.svc.DescribeSpotInstanceRequests(&ec2.DescribeSpotInstanceRequestsInput{})
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		for _, request := range resp.SpotInstanceRequests {
+			label := prometheus.Labels{
+				"account":           service.Account,
+				"region":            service.Region,
+				"state":             *request.State,
+				"code":              *request.Status.Code,
+				"instance_type":     *request.LaunchSpecification.InstanceType,
+				"instance_id":       *request.InstanceId,
+				"availability_zone": *request.LaunchedAvailabilityZone,
+			}
+			spotRequestItems = append(spotRequestItems, prom.SpotRequestItem{
+				Labels: label,
+				Count:  1,
```

Comment:
You actually do not need this `Count`, right? So `spotRequestItems` could actually be of the type `[]prometheus.Labels`.

Reply:
yea i guess i was saving that for some future purpose that will never come up probably. ;)